src/internal/stream.rs

```diff
@@ -299,12 +320,14 @@ fn read_data_from_stream<F: Read + Seek>(
         if stream_len < consts::MINI_STREAM_CUTOFF as u64 {
             let mut chain = minialloc.open_mini_chain(start_sector)?;
             chain.seek(SeekFrom::Start(buf_offset_from_start))?;
-            chain.read_exact(&mut buf[..num_bytes])?;
+            return Ok(read_until_error(&mut chain, &mut buf[..num_bytes])
```

Comment:
I'm a little confused what's going on with this change.  Seems like it's changing to ignore IO errors?  Why is that desired?

Reply:
@mdsteele fat entries could be truncated or incompleted. `read_exact` emits ` Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }` in case of incomplete data